### PR TITLE
Upgrade Boost to 1.70

### LINKED ISSUE #5614: Upgrade Boost to 1.70+
After Hunter is upgraded in #5506, I propose to use the latest Boost version. I noticed there are some API changes to Boost.ASIO. Someone will have to replace deprecated or remove API with a new one.

### PR DESCRIPTION
**Pull request by gumb0**
Fix #5614

Review commits separately.

### CODE REVIEW

**Comment by gumb0** [comment]
Windows build runs out of time, I'll check it on a local machine.

File: libp2p/Host.cpp [hunk 1]
```diff
@@ -508,21 +508,18 @@ void Host::runAcceptor()
         cnetdetails << "Listening on local port " << m_listenPort;
         m_accepting = true;
 
-        auto socket = make_shared<RLPXSocket>(m_ioContext);
-        m_tcp4Acceptor.async_accept(socket->ref(), [=](boost::system::error_code ec)
-        {
+        m_tcp4Acceptor.async_accept([=](boost::system::error_code _ec, bi::tcp::socket _socket) {
```
Comment (chfast) [inline libp2p/Host.cpp]: Does it pass it by value?

Comment (chfast) [reply libp2p/Host.cpp]: What other objects are captured? Can you be more specific than `=`?

Comment (gumb0) [reply libp2p/Host.cpp]: The documentation specifies the handler as `void handler(const boost::system::error_code& error, 
  typename Protocol::socket peer);`
https://www.boost.org/doc/libs/1_70_0/doc/html/boost_asio/reference/basic_socket_acceptor/async_accept/overload3.html

Looking in the debugger, it looks like it passes there `static_cast<socket&&>(peer)`

Comment (gumb0) [reply libp2p/Host.cpp]: Capturing `this` is enough, I'll change it

Comment (chfast) [reply libp2p/Host.cpp]: Ok, this looks like perfect forwarding so passing by reference will work as well. But I believe socket only has move constructor so we are fine here.

**Pull request merged**